### PR TITLE
Implement updateFinalPointsWithExternalTimestamps(), cleanup old setPoints(), fix GPX parsing of time with fractional seconds

### PR DESCRIPTION
**Pull request by RZR-UA**

### CODE REVIEW

**Comment by RZR-UA** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2342

**Comment by vshcherb** [comment]
4.7 branch!

File: OsmAnd-java/src/main/java/net/osmand/router/GpxSegmentsApproximation.java [hunk 1]
```diff
@@ -185,4 +186,75 @@ private void initGpxPointsXY31(List<GpxPoint> gpxPoints) {
 			p.y31 = MapUtils.get31TileNumberY(p.loc.getLatitude());
 		}
 	}
+
+	public static void updateFinalPointsWithExternalTimestamps(List<GpxPoint> finalPoints,
```
[the remark this answers is not in the document]
Comment (RZR-UA) [reply OsmAnd-java/src/main/java/net/osmand/router/GpxSegmentsApproximation.java:190]: Will be discussed and refactored if necessary.

**Comment by RZR-UA** [comment]
rebased to r4.7

**Pull request closed without merging**